File: backend/src/main/java/com/mycompany/contactmgr/dto/BusinessUnitDto.java
package com.mycompany.contactmgr.dto;

import java.time.Instant;
import java.util.UUID;
import com.mycompany.contactmgr.model.BusinessUnitStatus;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class BusinessUnitDto {
    private UUID businessUnitId;
    private String name;
    private String slug;
    private String description;
    private BusinessUnitStatus status;
    private Instant createdAt;
    private Instant updatedAt;

    // Constructor, Getters, and Setters (Omitted for brevity)
}

File: frontend/src/common/dto/BusinessUnitDto.ts
export type BusinessUnitStatus = 'ACTIVE' | 'INACTIVE' | 'PENDING';

export interface BusinessUnitDto {
  businessUnitId: string;       // UUID - Primary Identifier
  name: string;                 // e.g., "European Operations"
  slug: string;                 // URL-friendly identifier (e.g., "eu-operations")
  description: string;          // Full description of the BU's scope
  status: BusinessUnitStatus;   // ACTIVE, INACTIVE, PENDING
  createdAt: string;            // ISO Date String
  updatedAt: string;            // ISO Date String
}

export interface BusinessUnitSearchResponse {
  content: BusinessUnitDto[];
  number: number;
  size: number;
  totalPages: number;
  totalElements: number;
}

// Input DTO used when creating/updating a BU
export interface BusinessUnitInputDto {
  name: string;
  description: string;
  // Status is usually set by the service layer, but we allow override here.
  initialStatus: BusinessUnitStatus;
}
